sdk/core/azure_core_amqp/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp sasl
+#[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+mod fe2o3;
+
+#[cfg(any(not(feature = "enable-fe2o3-amqp"), target_arch = "wasm32"))]
+mod noop;
+
+pub mod cbs;
+pub mod connection;
+pub mod error;
+pub mod management;
+pub mod messaging;
+pub mod receiver;
+pub mod sender;
+pub mod session;
+pub mod value;
+
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SenderSettleMode {
+    Unsettled = 0,
+    Settled = 1,
+    Mixed = 2,
```

Comment:
Do these need values? This isn't normal. Really only useful if you're marshaling to/from another ABI e.g., C, or have to de/serialize these values. And in the former case, you should mark it as `#[repr(transparent)]` or `#[repr(C)]` - there's a subtle different I forget off hand but you can look it up if needed.

Reply:
Those values are defined by the [AMQP protocol](https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-sender-settle-mode). They are initialized to ensure that the internal representation matches the protocol.

I'm not comfortable using `#[repr(transparent)]` or `#[repr(C)]` because these are protocol elements - I would be initializing them in C/C++ as well.

Reply:
I've defined manifest constants for these values and then included a link to the protocol documentation which defines them.

Btw, I'm going to end up moving them out of the main crate, they really don't belong there.